python/mxnet/merger.py

```diff
@@ -0,0 +1,121 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# coding: utf-8
+# pylint: disable=too-many-lines
+"""Gradient merging functions."""
+
+
+class Merger(object):
+    """Merger for kvstore."""
```

Comment:
Need much more documentation. Why isn't there a constructor?
BTW this should be put under mxnet.kvstore

Reply:
> Why isn't there a constructor?

It's just a wrapper for a callback. Unlike `Updater` which maintains state, `Merger`, by default, is stateless.